src/main/java/org/primefaces/model/menu/DefaultMenuColumn.java

```diff
@@ -37,6 +37,9 @@
     private List<MenuElement> elements;
     private boolean rendered = true;
 
+    /**
+     * @deprecated use DefaultMenuColumn#Builder instead
+     */
```

Comment:
First, because I believe what is written in Effective Java is right and second, forcing does not mean "no using constructor is bad", it means "whatever you want to dp, you might want to you use the builder pattern :wink:"

Though, if by any chance building a default `DefaultMenuItem` (without any initialization) would make sense, I would remove the `@Deprecated` on no args constructor.

Reply:
Yep no problem.  The only thought of leaving the no arg constructor is if people are somehow producing and injecting these using CDI it will require a no arg constructor to be proxied.

Reply:
That would be a very strange way to create menus, but yes that's true